### ISSUE TITLE
Intellisense re-evaluation seems to be too frequent

### ISSUE THREAD

**Issue by knetworx** [BowlerHatLLC/vscode-nextgenas#205]
Intellisense seems to re-evaluate immediately anytime a single character changes. For example, I was replacing a variable in a function call, so I hit backspace about 10 times, then typed the other variable name. After I was done, I watched over the next 20-30 seconds as the Intellisense red squiggly went from the full length of the previous variable, down to one character, then back up until it reached the full length of the new variable, and went away. I don't know how the Intellisense code is implemented, and have never written IDE code myself, but from what I can tell in other editors, there's a sort of grace period when you change text, so that it doesn't re-evaluate Intellisense until about a second after the last character you type. Would that be possible to implement in this plugin?

**Comment by neminovno** [comment]
If I understand, it never happened to me. Maybe this setting can help:
```
// Controls the delay in ms after which quick suggestions will show up.
"editor.quickSuggestionsDelay": 10,
```

**Comment by knetworx** [comment]
Unfortunately, no, that's a different setting. That determines when it shows a completion suggestion - the issue I'm having is when it does validation (though you would think the two would be related). I tried setting it to 500 just in case, but I still got the same result.

**Comment by joshtynjala** [comment]
I'll try it with the mouse over tomorrow. That seems like it might be what I was missing.

**Comment by joshtynjala** [comment]
I randomly discovered that I'm able to consistently reproduce this issue in a huge file that has 8500+ lines of code. How many lines are in your file?

**Comment by knetworx** [comment]
The original file I noticed it in has about 1900, but it's also only one file of a few thousand.

**Comment by joshtynjala** [comment]
I've found a solution for this issue. It's going to take some work to come up with a robust solution, but I was able to speed things up significantly in a quick and dirty prototype.

**Comment by knetworx** [comment]
Awesome!

**Comment by joshtynjala** [comment]
Closed in commit 898bbe0a612c36401242ebb6f93cbbead6df49d4
